packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -28,7 +28,7 @@ import { addQueryArgs } from '@wordpress/url';
  * Module constants
  */
 const DEFAULT_QUERY = {
-	per_page: -1,
+	per_page: 20,
```

Comment:
What happens if I type 3 characters and have 25 results?

Reply:
IMO if this should change ( I believe it shouldn't and be handled in lower lever ), it should be at least the previous API limit (100 ?) for a quicker `logical?` enhancement/fix.

Also a limit of input characters for triggering a request should exist regardless of the `per_page` value.

Reply:
I set 20 to stay consistent (consequent) with the maxSuggestion previously set.
I don't mind having 100, but we are not adding it to the suggestions anyway, so why adding more? It makes the query quicker to have the lowest limit used.

The limit to character is more of a UX point. What I am trying to fix here is the number of AJAX call triggered for a simple search when you have a ton of tags.

Reply:
So if we have a MAX_TERMS_SUGGESTIONS constant we should use that constant's value to set the size of the response we expect. It's, indeed, no point in getting 2293 results and only display 20.

so probably something like:
```
const MAX_TERMS_SUGGESTIONS = 20;
const DEFAULT_QUERY = {
	per_page: MAX_TERMS_SUGGESTIONS,
	per_page: 20,
....
```

What is the default sort here?

Reply:
Count is, defined on the line right after this targeted comment line.

`orderby: 'count',
	order: 'desc',
	_fields: 'id,name',`

that means, the more used is the first, since it is the most likely if we take a stat approach.

Reply:
> So if we have a MAX_TERMS_SUGGESTIONS constant we should use that constant's value to set the size of the response we expect.

This sounds right. @draganescu 's comment is good, just add only `per_page: MAX_TERMS_SUGGESTIONS,`